sdk/core/core/src/az_span.c

```diff
@@ -139,8 +139,8 @@ AZ_NODISCARD az_result az_span_to_uint32(az_span self, uint32_t* out)
  */
 AZ_NODISCARD az_result az_span_copy(az_span dst, az_span src, az_span* out)
 {
-  AZ_PRECONDITION_VALID_SPAN(dst, 0);
-  AZ_PRECONDITION_VALID_SPAN(src, 0);
+  AZ_PRECONDITION_VALID_SPAN(dst, 0, false);
+  AZ_PRECONDITION_VALID_SPAN(src, 0, false);
```

Comment:
NULL is allowed for both of these; change to true.
Copying to/from NULL is like a no-op. It's like having a null stream where you can stream to null to just consume data. Or like memcopy from a buffer with 0 bytes - this is OK